File: Model/src/main/java/com/td/model/entity/dictionary/user/UserModel_.java
package com.td.model.entity.dictionary.user;

import com.td.model.entity.AbstractModel_;
import com.td.model.entity.dictionary.Person;

import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

/**
 * Created by konstantinchipunov on 25.07.14.
 */
@StaticMetamodel(UserModel.class)
public abstract class UserModel_ extends AbstractModel_ {

    public static volatile SingularAttribute<UserModel, String> phone;
    public static volatile SingularAttribute<UserModel, String> mail;
    public static volatile SingularAttribute<UserModel, String> login;
    public static volatile SingularAttribute<UserModel, Person.Gender> gender;
    public static volatile SingularAttribute<UserModel, String> patronymic;
    public static volatile SingularAttribute<UserModel, String> lastName;
    public static volatile SingularAttribute<UserModel, String> firstName;

}

File: WebApp/src/main/webapp/app/model/dictionary/UserModel.js
Ext.define('TransDocs.model.dictionary.UserModel', {
    extend: 'TransDocs.model.AbstractModel',
    entityName: "User",
    fields:[
        {name: 'mail', type: 'string'},
        {name: 'phone', type: 'string'},
        {name: 'login', type: 'string'},
        {name: 'gender', type: 'string'},
        {name: 'patronymic', type: 'string'},
        {name: 'lastName', type: 'string'},
        {name: 'firstName', type: 'string'},
        {name: 'description', type: 'string'},
        {
            name: 'companyId', reference: {
            type: 'Company',
            association: 'CompanyModelByUserModel',
            role: 'company',
            inverse: 'users'
        }
        },
        {name: 'leaf', type:"boolean", convert: function(value, record){
            return record.get("objectId")!=record.getObjectType();
        }, persist:false}
    ],

    requires: [
        'TransDocs.model.dictionary.RoleModel',
        "TransDocs.model.dictionary.CompanyModel",
        'TransDocs.data.reader.DefaultJsonReader'
    ],

    manyToMany: {
        roleModels: {
            type: 'TransDocs.model.dictionary.RoleModel',
            role: 'roleModels',
            field: 'objectId'
        }
    },


    getObjectType: function(){
        return "td_user";
    }

});